lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -66,6 +66,23 @@ internal class EvaluatingCompiler(
         get() = compilationContextStack.peek() ?: throw EvaluationException(
             "compilationContextStack was empty.", internal = true)
 
+
+    /**
+     * This checks whether the thread has been interrupted. Specifically, it currently checks during the compilation
+     * of aggregations and joins, the "evaluation" of aggregations and joins, and the materialization of joins
+     * and from source scans.
+     *
+     * Note: This is essentially a way to avoid constantly checking [CompileOptions.interruptible]. By writing it this
+     * way, we statically determine whether to introduce checks. If the compiler has specified
+     * [CompileOptions.interruptible], the invocation of this function will insert a Thread interruption check. If not
+     * specified, it will not perform the check during compilation/evaluation/materialization.
+     */
+    private val interruptionCheck: () -> Unit = {
+            if (Thread.interrupted()) {
```

Comment:
Should this be behind a `CompileOptions` flag as the comment mentions? The interruption support added to `main` in #1211 puts the interruption check behind a compile option flag.

Reply:
Agreed. I just pushed an update to add this. I originally wasn't sure due to adding new APIs in a patch. I decided to add a deprecated notice on the API.